ui/page/dexclient/orders_history_page.go

```diff
@@ -0,0 +1,191 @@
+package dexclient
+
+import (
+	"context"
+	"fmt"
+
+	"decred.org/dcrdex/client/core"
+	"gioui.org/layout"
+	"gioui.org/widget"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/load"
+	"github.com/planetdecred/godcr/ui/page/components"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+const OrdersHistoryPageID = "OrdersHistory"
+
+type OrdersHistoryPage struct {
+	*load.Load
+	ctx        context.Context
+	ctxCancel  context.CancelFunc
+	list       *widget.List
+	backButton decredmaterial.IconButton
+	host       string
+	orders     []*core.Order
+}
+
+func NewOrdersHistoryPage(l *load.Load, host string) *OrdersHistoryPage {
+	pg := &OrdersHistoryPage{
+		Load: l,
+		host: host,
+		list: &widget.List{
+			List: layout.List{Axis: layout.Vertical},
+		},
+	}
+
+	pg.backButton, _ = components.SubpageHeaderButtons(pg.Load)
+
+	return pg
+}
+
+func (pg *OrdersHistoryPage) ID() string {
+	return OrdersHistoryPageID
+}
+
+func (pg *OrdersHistoryPage) OnResume() {
+	pg.ctx, pg.ctxCancel = context.WithCancel(context.TODO())
+	go pg.readNotifications()
+	pg.getOrdersHistory()
+}
+
+func (pg *OrdersHistoryPage) Layout(gtx layout.Context) layout.Dimensions {
+	body := func(gtx C) D {
+		sp := components.SubPage{
+			Load:       pg.Load,
+			Title:      "Orders History",
+			BackButton: pg.backButton,
+			Back: func() {
+				pg.PopFragment()
+			},
+			Body: func(gtx layout.Context) layout.Dimensions {
+				return layout.Inset{
+					Left:  values.MarginPadding10,
+					Right: values.MarginPadding10,
+				}.Layout(gtx, func(gtx C) D {
+					return pg.pageSections(gtx, func(gtx C) D {
+						gtx.Constraints.Min.Y = gtx.Constraints.Max.Y
+						return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+							layout.Rigid(func(gtx C) D {
+								return layout.Flex{}.Layout(gtx,
+									layout.Flexed(0.125, func(gtx C) D {
+										return pg.Theme.Label(values.TextSize14, "Trade").Layout(gtx)
+									}),
+									layout.Flexed(0.125, func(gtx C) D {
+										return pg.Theme.Label(values.TextSize14, "Side").Layout(gtx)
+									}),
+									layout.Flexed(0.125, func(gtx C) D {
+										return pg.Theme.Label(values.TextSize14, "Rate").Layout(gtx)
+									}),
+									layout.Flexed(0.125, func(gtx C) D {
```

Comment:
Instead of having to repeat same label multiple time, you can add this to a function.
```
func (pg *OrdersHistoryPage) layoutRows(text string) layout.Widget {
	return pg.Theme.Label(values.TextSize14, text).Layout
}
```

Reply:
Or `layoutLabel`. It really is laying out a label, although the caller in this instance is using the label as a row, which is fine, but the method lays out a label.